### ISSUE TITLE
PolynomialSamplingCheck failing test case

### ISSUE THREAD

**Issue by tixxit** [non/spire#608]
This tests failed: https://github.com/non/spire/blob/master/tests/src/test/scala/spire/math/PolynomialSamplingCheck.scala#L84

With the following case: 

```
- rational/sparse binop /~ and % *** FAILED *** (213 milliseconds)
   ArithmeticException was thrown during property evaluation.
     Message: divide (/) by 0
     Occurred when passed generated values (
       arg0 = (-21/2x⁹⁴),
       arg1 = (1/84x⁵ - 949/448x + 1),
       arg2 = -55/53
     )
```

We should fix this.

**Comment by denisrosset** [comment]
Here is a new one.
```
[info] - rational/sparse (x /~ y) * y + (x % y) = x *** FAILED *** (750 milliseconds)
[info]   ArithmeticException was thrown during property evaluation.
[info]     Message: divide (/) by 0
[info]     Occurred when passed generated values (
[info]       arg0 = (8x⁹⁹ - 8x⁹¹ - 8/29x⁹⁰ + 8x⁶⁷ - 8/13x⁶⁵ + 75x⁶⁴ - 527/56x⁴⁸ - 64685/2112x⁴⁷ - 92/55x³⁹ + 88/9x³⁰ + 2x²⁶ + 99/58x⁹ - 105643/840x),
[info]       arg1 = (-40/11x⁹⁶ - 1/60x⁷⁹ - 9/8x⁷³ - 1/7x⁷² - 30/49x⁶² - 11/3x⁴⁹ - 20247/353800x⁴⁸ - 5746/49x⁴⁷ + 8x³² - 60x³¹ - x¹³ + 2/3x⁹ + 55603379/218592x - 57227/840)
[info]     )
```

**Comment by SethTisue** [comment]
seen in Scala 2.12 community build: https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1456/consoleFull

```
[spire] [info] - rational/sparse p + 0 = p *** FAILED *** (488 milliseconds)
[spire] [info]   TestFailedException was thrown during property evaluation.
[spire] [info]     Message: (-9x⁸⁵ + 85/49x⁸⁰ + 9/10x⁷⁵ - 4/9x⁷² + 90x⁶² + 1/7x⁶¹ - x⁵⁶ - 1/7x⁵³ - 18/7x⁵² - 1238661/14938x⁴⁸ + 16776521/116622x⁴⁷ - 79/8x⁴⁰ - 62/7x²⁹ - x²⁵ - 8x²² - 7/2x²⁰ + 82x¹² - 659/518x¹⁰ + 13/7x³ - 4871536333/167785828x + 382111/6426) was not equal to (-9x⁸⁵ + 85/49x⁸⁰ + 9/10x⁷⁵ - 4/9x⁷² + 90x⁶² + 1/7x⁶¹ - x⁵⁶ - 1/7x⁵³ - 18/7x⁵² - 1238661/14938x⁴⁸ + 16776521/116622x⁴⁷ - 79/8x⁴⁰ - 62/7x²⁹ - x²⁵ - 8x²² - 7/2x²⁰ + 82x¹² - 659/518x¹⁰ + 13/7x³ - 4871536333/167785828x + 382111/6426)
[spire] [info]     Location: (PolynomialTest.scala:97)
[spire] [info]     Occurred when passed generated values (
[spire] [info]       arg0 = (-9x⁸⁵ + 85/49x⁸⁰ + 9/10x⁷⁵ - 4/9x⁷² + 90x⁶² + 1/7x⁶¹ - x⁵⁶ - 1/7x⁵³ - 18/7x⁵² - 1238661/14938x⁴⁸ + 16776521/116622x⁴⁷ - 79/8x⁴⁰ - 62/7x²⁹ - x²⁵ - 8x²² - 7/2x²⁰ + 82x¹² - 659/518x¹⁰ + 13/7x³ - 4871536333/167785828x + 382111/6426)
[spire] [info]     )
[spire] [info]   org.scalatest.exceptions.GeneratorDrivenPropertyCheckFailedException:
[spire] [info]   at org.scalatest.enablers.CheckerAsserting$$anon$2.indicateFailure(CheckerAsserting.scala:232)
```

**Comment by denisrosset** [comment]
Should be addressed by #653
